backend/public/index.html

```diff
@@ -1,7 +1,145 @@
+            <button id="googleLogin">Đăng nhập bằng Google</button>
+            <div id="result" class="result" style="display:none"></div>
+        </div>
+    </div>
+
+    <script>
+        const CLIENT_ID = '816422652963-euhfse499nkf8mvfb18t3m1hdroqo7p5.apps.googleusercontent.com';
+
+        function rnd() { return Math.random().toString(36).slice(2) + Date.now().toString(36); }
+        function buildAuthUrl() {
+            const url = new URL('https://accounts.google.com/o/oauth2/v2/auth');
+            url.searchParams.set('client_id', CLIENT_ID);
+            url.searchParams.set('redirect_uri', 'http://localhost:5500/backend/public');
+            url.searchParams.set('response_type', 'token id_token');
+            url.searchParams.set('scope', 'openid email profile');
+            url.searchParams.set('nonce', rnd());
+            url.searchParams.set('state', rnd());
+            url.searchParams.set('include_granted_scopes', 'true');
+            url.searchParams.set('prompt', 'select_account');
+            return url.toString();
+        }
+
+        function parseHash() {
+            if (!location.hash.startsWith('#')) return {};
+            const p = new URLSearchParams(location.hash.slice(1));
+            return {
+                accessToken: p.get('access_token') || '',
+                idToken: p.get('id_token') || '',
+                expiresIn: p.get('expires_in') || '',
+                tokenType: p.get('token_type') || '',
+                scope: p.get('scope') || ''
+            };
+        }
+
+        async function postToBackend(tokens) {
+            const url = 'http://localhost:3000/api/v1/auths/google';
+            try {
+                const res = await fetch(url, {
+                    method: 'POST',
+                    headers: { 'Content-Type': 'application/json' },
+                    body: JSON.stringify({ idToken: tokens.idToken, accessToken: tokens.accessToken }),
+                });
+                const text = await res.text();
+                try {
+                    return { status: res.status, body: JSON.parse(text) };
+                } catch { return { status: res.status, body: text } }
+            } catch (e) {
+                return { status: 0, body: String(e) };
+            }
+        }
+
+        function showResult(tokens, backendResp) {
+            const el = document.getElementById('result');
+            el.style.display = 'block';
+            el.innerHTML = '<div><strong>Backend response (status ' + backendResp.status + ')</strong>:</div>'
+                + '<pre>' + (typeof backendResp.body === 'string' ? backendResp.body : JSON.stringify(backendResp.body, null, 2)) + '</pre>';
+        }
+
+        document.getElementById('googleLogin').addEventListener('click', () => {
+            location.assign(buildAuthUrl());
+        });
+
+        // If redirected back with tokens in hash, show them and clean URL
+        (async function handleReturn() {
```

Comment:
Nice to have: 

- Hiển thị tên của user đang đăng nhập bằng cách gọi api getMe
- Refresh token khi token hết hạn